test_conformance/api/test_clone_kernel.cpp

```diff
@@ -351,51 +466,468 @@ int test_clone_kernel(cl_device_id deviceID, cl_context context, cl_command_queu
+        "argument.");
+
+    test_assert_error(
+        ((float*)pbufRes)[2] == farg,
+        "clCloneKernel test failed. Failed to clone float type argument.");
+
+    test_assert_error(
+        ((float*)pbufRes)[3] == sa.f,
+        "clCloneKernel test failed. Failed to clone structure type "
+        "argument.");
+
+    test_assert_error(
+        ((int*)pbufRes)[6] == write_val,
+        "clCloneKernel test failed.  Failed to clone cl_mem argument.");
+
+    if (bimg)
     {
-        test_error( error, "clCloneKernel test failed. Failed to clone integer type argument." );
-        return -1;
+        error = test_image_arg_shallow_clone(deviceID, context, queue,
+                                             num_elements, pbufRes, bufOut);
+        test_error(error, "image arg shallow clone test failed.");
     }
 
-    if (((int*)pbufRes)[1] != sa.i)
+    if (bdouble)
     {
-        test_error( error, "clCloneKernel test failed. Failed to clone structure type argument." );
-        return -1;
+        error = test_double_arg_clone(deviceID, context, queue, num_elements,
+                                      pbufRes, bufOut);
+        test_error(error, "double arg clone test failed.");
+    }
+
+    delete[] pbuf;
+    delete[] pbufRes;
+
+    return TEST_PASS;
+}
+
+int test_exec_enqueue_helper(cl_context context, cl_command_queue queue,
+                             BufPtr* pBuf, cl_int* svmPtr_Kernel,
+                             cl_kernel* srcKernel, cl_int* value)
+{
+    cl_int error;
+    size_t ndrange1 = 1;
+
+    // enqueue - srcKernel
+    error = clEnqueueNDRangeKernel(queue, *srcKernel, 1, NULL, &ndrange1, NULL,
+                                   0, NULL, NULL);
+    test_error(error, "clEnqueueNDRangeKernel failed");
+    error = clFinish(queue);
+
+    test_error(error, "clFinish failed");
+    error = clEnqueueSVMMap(queue, CL_TRUE, CL_MAP_READ | CL_MAP_WRITE,
+                            svmPtr_Kernel, sizeof(cl_int), 0, NULL, NULL);
+    test_error(error, "clEnqueueSVMMap failed");
+    test_assert_error(svmPtr_Kernel[0] == *value,
+                      "clCloneKernel test failed, Failed to verify "
+                      "integer value from SVM pointer. ");
+
+    error = clEnqueueSVMUnmap(queue, svmPtr_Kernel, 0, NULL, NULL);
+    test_error(error, "clEnqueueSVMUnmap failed");
+    error = clFinish(queue);
+    test_error(error, "clFinish failed");
+
+    return TEST_PASS;
+}
+
+int test_svm_exec_info_helper(cl_context context, cl_command_queue queue,
+                              BufPtr* pBuf, cl_int* svmPtr_Kernel,
+                              cl_kernel* srcKernel, cl_int* value)
+{
+    cl_int error;
+
+    error = clEnqueueSVMMap(queue, CL_TRUE, CL_MAP_READ | CL_MAP_WRITE, pBuf,
+                            sizeof(BufPtr), 0, NULL, NULL);
+    test_error(error, "clEnqueueSVMMap failed");
+    error = clFinish(queue);
+    test_error(error, "clFinish failed");
```

Comment:
This `clFinish` should not be needed, since `clEnqueueSVMMap` is made blocking by passing it `CL_TRUE` as the second argument.

Reply:
done